addons/purchase/stock.py

```diff
@@ -151,14 +151,73 @@ def _get_picking_to_recompute(self, cr, uid, ids, context=None):
                 picking_ids.add(move.picking_id.id)
         return list(picking_ids)
 
+    def _get_purchase_id(self, cr, uid, ids, name, args, context=None):
+        pur_obj = self.pool.get("purchase.order")
+        res = {}
+        for picking in self.browse(cr, uid, ids, context=context):
+            res[picking.id] = False
+            if picking.group_id:
+                # TODO: Check why group_id is not assigned for purchase order, while sale order works fine (while fetching through procurement order)
+                # purchase_ids = pur_obj.search(cr, uid, [('group_id', '=', picking.group_id.id)], context=context)
+                purchase_ids = pur_obj.search(cr, uid, [('name', '=', picking.group_id.name)], context=context)
```

Comment:
we cannot search on 'name' field. name can be changed.
can not possible read_group or sql query there is picking_ids one2many field.?